### PR TITLE
Potentially Fixes keycloak/keycloak#25731 Fetch group count of subgroups at query time to avoid N+1 database queries when compiling response

### PR DESCRIPTION
**Pull request by alice-wondered**
This doesn't resolve the double query of group ids but this does prevent N+1 queries back to the DB for the count value. Should at least reduce the impact of adding the count field. 

A more direct approach would be to add this count field to all relevant named queries and have those named queries return the entities directly rather than IDs. Cache would have to be modified to work correctly in this case

### CODE REVIEW

File: model/jpa/src/main/resources/META-INF/jpa-changelog-24.0.0.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  ~ * Unless required by applicable law or agreed to in writing, software
+  ~ * distributed under the License is distributed on an "AS IS" BASIS,
+  ~ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ * See the License for the specific language governing permissions and
+  ~ * limitations under the License.
+  -->
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
+
+    <!-- I'm not sure how these change set numbers are calculated -->
+    <changeSet author="keycloak" id="24.0.0-12062">
```
Comment (alice-wondered) [inline model/jpa/src/main/resources/META-INF/jpa-changelog-24.0.0.xml:21]: I'm not sure what 12062 refers to, just copied off of the last file

**Comment by pedroigor** [comment]
@alice-wondered We just merged https://github.com/keycloak/keycloak/pull/26904. It should provide a fix to the original problem without introducing:

* A new database index and potentially impact migration. We also need some benchmark to understand its impact. If you are still interested, we can evaluate this as a follow-up.
* Assume that the sub-group count should be always fetched. Not sure if we should be so opinionated about returning the subgroup count every time we are fetching a group by ID. I also think it deserves a follow-up to see how we can improve cases where we want groups and subgroups within the same transaction.

I'm closing the PR for these reasons.

**Comment by michbeck100** [comment]
I would like to see a follow-up of this. The very first call to getGroups is still slow because of the N+1 selects. @alice-wondered would be great if you would continue to work on this!

**Comment by alice-wondered** [comment]
> * Assume that the sub-group count should be always fetched. Not sure if we should be so opinionated about returning the subgroup count every time we are fetching a group by ID. I also think it deserves a follow-up to see how we can improve cases where we want groups and subgroups within the same transaction.

 I go back and forth on how I feel about this. I originally went with the less opinionated approach and that seems to have caused a lot of issues for people which is why I'm thinking the more performant opinion could be better for that. 
 
 Definitely interested in following up on these issues though. The database index is probably something that is a minor optimization but with how many operations read that column could avoid quite a few table scans.

**Pull request closed without merging**